presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetTypeUtils.java

```diff
@@ -183,22 +191,33 @@ public static Type getPrestoType(TupleDomain<ColumnDescriptor> effectivePredicat
         }
     }
 
-    private static Type createVarcharType(TupleDomain<ColumnDescriptor> effectivePredicate, RichColumnDescriptor column)
+    private static Type createVarcharType(RichColumnDescriptor descriptor)
     {
-        // We look at the effectivePredicate domain here, because it matches the Hive column type
-        // more accurately than the type available in the RichColumnDescriptor.
-        // For example, a Hive column of type varchar(length) is encoded as a Parquet BINARY, but
-        // when that is converted to a Presto Type the length information wasn't retained.
-        Optional<Map<ColumnDescriptor, Domain>> predicateDomains = effectivePredicate.getDomains();
-        if (predicateDomains.isPresent()) {
-            Domain domain = predicateDomains.get().get(column);
-            if (domain != null) {
-                return domain.getType();
+        if (descriptor.getHiveType().isPresent()) {
+            TypeInfo typeInfo = descriptor.getHiveType().get().getTypeInfo();
+            if (typeInfo instanceof VarcharTypeInfo) {
+                return VarcharType.createVarcharType(((VarcharTypeInfo) typeInfo).getLength());
             }
         }
         return VarcharType.VARCHAR;
     }
 
+    private static Type createIntegerType(RichColumnDescriptor descriptor)
+    {
+        if (descriptor.getHiveType().isPresent()) {
+            TypeInfo typeInfo = descriptor.getHiveType().get().getTypeInfo();
+            switch(typeInfo.getTypeName()){
+                case StandardTypes.SMALLINT:
```

Comment:
Does `tinyint` need to be handled here? In https://github.com/prestodb/presto/issues/8243#issue-235274112 they mention getting the error `Mismatched Domain types: tinyint vs integer`. 

Also, maybe verify dictionary filtering works for these smaller int types. It looks like min/max filtering would work [here](https://github.com/prestodb/presto/blob/master/presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/TupleDomainParquetPredicate.java#L147) but dictionary filtering doesn't check for those types e.g. [here](https://github.com/prestodb/presto/blob/master/presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/TupleDomainParquetPredicate.java#L238) although that's probably less common of a use case.